File: frontend/src/api/permission.ts
import request from '@/utils/request';
import type {
  CreatePermissionParams,
  UpdatePermissionParams,
  PermissionInfo,
  PermissionTreeResult,
} from '@/types/permission';

export function createPermission(data: CreatePermissionParams) {
  return request.post('/permissions', data);
}

export function updatePermission(id: number, data: UpdatePermissionParams) {
  return request.put(`/permissions/${id}`, data);
}

export function deletePermission(id: number) {
  return request.delete(`/permissions/${id}`);
}

export function getPermissionTree() {
  return request.get<PermissionTreeResult>('/permissions');
}

export function getUserPermissions() {
  return request.get('/user/permissions');
}

File: routes/api/v1/permission.go
package v1

import (
	"datax-admin/controllers"
	"github.com/gin-gonic/gin"
)

// RegisterPermissionRoutes 注册权限管理相关路由
func RegisterPermissionRoutes(authenticated *gin.RouterGroup) {
	permissionController := controllers.NewPermissionController()
	
	authenticated.POST("/permissions", permissionController.CreatePermission)
	authenticated.PUT("/permissions/:id", permissionController.UpdatePermission)
	authenticated.DELETE("/permissions/:id", permissionController.DeletePermission)
	authenticated.GET("/permissions", permissionController.GetPermissionTree)
	authenticated.GET("/user/permissions", permissionController.GetUserPermissions)
}